neet/boolean/randomnet.py

```diff
@@ -66,7 +66,15 @@ def _random_binary_states(n, p):
     return set(tuple(state) for state in StateSpace(n) if random.random() < p)
 
 
-def _random_logic_fixed_connections(logic_net, ps):
+def _external_nodes(logic_net):
```

Comment:
This function defines a node as external if and only if it has exactly one incoming edge, that edge is a self-edge, and that self-edge represents the "identify" function (0 -> 0, 1 -> 1). Is this correct? My concern is whether or not we can have networks with nodes that meet this criterion, but aren't listed in the "external.txt" file as an external edge.

Reply:
I thought about this possibility when I wrote this. But the "external" nodes are stored in the LogicNetwork just as any other nodes. We have to give `LogicNetwork` an attribution `externals` and store external nodes explicitly to preserve the difference. Then we have to consider adding the same attribute the `WTNetwork` as there is no reason why `LogicNetwork` gets this special treatment. So before we do that, I convinced myself that when  they built the biological networks, any node that would have logic equation A = A is already put in the "external.txt" file. But I admit this assumption is likely wrong.

Reply:
I did a quick check. There are indeed some networks which have nodes with expression A = A but not included in the "expression.txt" file.